Satellite 5 scan is not inspecting all the systems

## Specify type:
- Bug

### Priority:
- High
___

## Description:
Checking the reports of a Satellite 5 scan it shows that the number of systems scanned is not the same of the system count. That means that the scan is not inspecting all systems found.

For example the below task_results are from a Satellite 5 scan:

```
E          'task_results': [{'end_time': '2018-05-04T18:11:36.421734',
E                            'scan_type': 'connect',
E                            'source': 5,
E                            'start_time': '2018-05-04T18:09:15.706629',
E                            'status': 'completed',
E                            'status_message': 'Task is complete.',
E                            'systems_count': 4206,
E                            'systems_failed': 0,
E                            'systems_scanned': 4206},
E                           {'end_time': '2018-05-04T18:29:33.531930',
E                            'scan_type': 'inspect',
E                            'source': 5,
E                            'start_time': '2018-05-04T18:11:36.438433',
E                            'status': 'failed',
E                            'status_message': 'Satellite error encountered: [Errno 110] '
E                                              'Connection timed out. Inspect scan '
E                                              'failed for '
E                                              '18d57a13-7cb1-4514-8456-b865c1a9a00b.',
E                            'systems_count': 4206,
E                            'systems_scanned': 1386}]}
```

___

## Acceptance Criteria:

- [ ] Verify that a Satellite 5 scan reports `systems_count == systems_scanned + systems_failed + systems_unreachable`

___

## Bug Report

### Version Information:
6738e2aa8d220463bbb7d966dd1903e4188c5b6c

### Expected behavior:
All found systems are inspected

### Actual behavior:
Not all found systems are inspected

### Steps to reproduce:
Perform a Satellite 5 scan and check the task_results.

### Environment information:
Provide detailed information on the *scanning* and *target* systems (if applicable); see example below.

Type            | Operating System | OS Version | Python Version | Virtualization
--------------- | ---------------- | ---------- | -------------- | --------------
Scanning System | Fedora           | 27         | 3.6            | Bare Metal
Target System   | Satellite        | 5          | N/A            | N/A

--- a/quipucords/api/scanjob/model.py
+++ b/quipucords/api/scanjob/model.py
@@ -149,7 +149,8 @@ class ScanJob(models.Model):
 
         :param connect_only: counts should only include
         connection scan results
-        :return: systems_count, systems_scanned, systems_failed
+        :return: systems_count, systems_scanned,
+        systems_failed, systems_unreachable
         """
         if self.status == ScanTask.CREATED or \
                 self.status == ScanTask.PENDING:
@@ -189,7 +190,8 @@ class ScanJob(models.Model):
     def _calculate_counts(self, scan_type):
         """Calculate scan counts from tasks.
 
-        :return: systems_count, systems_scanned, systems_failed
+        :return: systems_count, systems_scanned,
+        systems_failed, systems_unreachable
         """
         systems_count = 0
         systems_scanned = 0

--- a/quipucords/api/scantasks/model.py
+++ b/quipucords/api/scantasks/model.py
@@ -260,7 +260,8 @@ class ScanTask(models.Model):
     def calculate_counts(self):
         """Calculate scan counts for task.
 
-        :return: systems_count, systems_scanned, systems_failed
+        :return: systems_count, systems_scanned,
+        systems_failed, systems_unreachable
         """
         systems_count = 0
         systems_scanned = 0

--- a/quipucords/api/scantasks/model.py
+++ b/quipucords/api/scantasks/model.py
@@ -257,6 +257,37 @@ class ScanTask(models.Model):
                           sys_failed=sys_failed,
                           sys_unreachable=sys_unreachable)
 
+    def calculate_counts(self):
+        """Calculate scan counts for task.
+
+        :return: systems_count, systems_scanned, systems_failed
+        """
+        systems_count = 0
+        systems_scanned = 0
+        systems_failed = 0
+        systems_unreachable = 0
+        if self.systems_count is not None:
+            if systems_count is None:
+                systems_count = 0
+            systems_count += self.systems_count
+        if self.systems_scanned is not None:
+            if systems_scanned is None:
+                systems_scanned = 0
+            systems_scanned += self.systems_scanned
+        if self.systems_failed is not None:
+            if systems_failed is None:
+                systems_failed = 0
+            systems_failed += self.systems_failed
+        if self.systems_unreachable is not None:
+            if systems_unreachable is None:
+                systems_unreachable = 0
+            systems_unreachable += self.systems_unreachable
+
+        return systems_count,\
+            systems_scanned,\
+            systems_failed,\
+            systems_unreachable
+
     @transaction.atomic
     def start(self):
         """Start a task."""

--- a/quipucords/api/scantasks/tests_scantask.py
+++ b/quipucords/api/scantasks/tests_scantask.py
@@ -170,3 +170,24 @@ class ScanTaskTest(TestCase):
         self.assertEqual(2, task.systems_scanned)
         self.assertEqual(2, task.systems_failed)
         self.assertEqual(2, task.systems_unreachable)
+
+    def test_calculate_counts(self):
+        """Test calculate counts."""
+        task = ScanTask.objects.create(
+            source=self.source,
+            scan_type=ScanTask.SCAN_TYPE_CONNECT,
+            status=ScanTask.PENDING)
+        # pylint: disable=invalid-name
+        task.save()
+        task.increment_stats('foo', increment_sys_count=True,
+                             increment_sys_scanned=True,
+                             increment_sys_failed=True,
+                             increment_sys_unreachable=True)
+        systems_count,\
+            systems_scanned,\
+            systems_failed,\
+            systems_unreachable = task.calculate_counts()
+        self.assertEqual(systems_count, 1)
+        self.assertEqual(systems_scanned, 1)
+        self.assertEqual(systems_failed, 1)
+        self.assertEqual(systems_unreachable, 1)

--- a/quipucords/scanner/satellite/five.py
+++ b/quipucords/scanner/satellite/five.py
@@ -88,10 +88,13 @@ class SatelliteFive(SatelliteInterface):
             client.auth.logout(key)
 
             for system in systems:
-                name = system.get(NAME)
-                if name is not None:
-                    hosts.append(name)
-                    self.record_conn_result(name, credential)
+                host_name = system.get(NAME)
+                host_id = system.get(ID)
+
+                if host_name is not None and host_id is not None:
+                    unique_name = '%s_%s' % (host_name, host_id)
+                    hosts.append(unique_name)
+                    self.record_conn_result(unique_name, credential)
 
         except xmlrpc.client.Fault as xml_error:
             raise SatelliteException(str(xml_error))
@@ -116,8 +119,9 @@ class SatelliteFive(SatelliteInterface):
             raise SatelliteException(
                 'host_details cannot be called for a connection scan')
         details = {}
+        unique_name = '%s_%s' % (host_name, host_id)
         sys_result = self.inspect_scan_task.inspection_result.systems.filter(
-            name=host_name).first()
+            name=unique_name).first()
 
         if sys_result:
             logger.debug('Results already captured for host_name=%s',
@@ -125,7 +129,7 @@ class SatelliteFive(SatelliteInterface):
             return None
         client, user, password = utils.get_sat5_client(self.inspect_scan_task)
         try:
-            message = 'REQUESTING HOST DETAILS: %s' % host_name
+            message = 'REQUESTING HOST DETAILS: %s' % unique_name
             self.inspect_scan_task.log_message(message)
 
             key = client.auth.login(user, password)
@@ -203,13 +207,13 @@ class SatelliteFive(SatelliteInterface):
 
             logger.debug('host_id=%s, host_details=%s',
                          host_id, details)
-            return {'host_name': host_name,
+            return {'name': unique_name,
                     'details': details,
                     'status': SystemInspectionResult.SUCCESS}
         except xmlrpc.client.Fault as xml_error:
             error_message = 'Satellite error encountered: %s\n' % xml_error
             logger.error(error_message)
-            return {'host_name': host_name,
+            return {'name': unique_name,
                     'details': details,
                     'status': SystemInspectionResult.FAILED}
 
@@ -338,6 +342,8 @@ class SatelliteFive(SatelliteInterface):
                 for result in results:
                     if result is not None:
                         self.record_inspect_result(
-                            result.get('host_name'),
+                            result.get('name'),
                             result.get('details'),
                             result.get('status'))
+
+        utils.validate_task_stats(self.inspect_scan_task)

--- a/quipucords/scanner/satellite/six.py
+++ b/quipucords/scanner/satellite/six.py
@@ -389,10 +389,13 @@ class SatelliteSixV1(SatelliteInterface):
                                              (response.status_code, url))
                 jsonresult = response.json()
                 for result in jsonresult.get(RESULTS, []):
-                    name = result.get(NAME)
-                    if name is not None:
-                        hosts.append(name)
-                        self.record_conn_result(name, credential)
+                    host_name = result.get(NAME)
+                    host_id = result.get(ID)
+
+                    if host_name is not None and host_id is not None:
+                        unique_name = '%s_%s' % (host_name, host_id)
+                        hosts.append(unique_name)
+                        self.record_conn_result(unique_name, credential)
 
         return hosts
 
@@ -408,8 +411,9 @@ class SatelliteSixV1(SatelliteInterface):
             raise SatelliteException(
                 'host_details cannot be called for a connection scan')
         details = {}
+        unique_name = '%s_%s' % (host_name, host_id)
         sys_result = self.inspect_scan_task.inspection_result.systems.filter(
-            name=host_name).first()
+            name=unique_name).first()
 
         if sys_result:
             logger.debug('Results already captured for host_name=%s',
@@ -417,7 +421,7 @@ class SatelliteSixV1(SatelliteInterface):
             return None
 
         try:
-            message = 'REQUESTING HOST DETAILS: %s' % host_name
+            message = 'REQUESTING HOST DETAILS: %s' % unique_name
             self.inspect_scan_task.log_message(message)
             details.update(host_fields(self.inspect_scan_task, 1,
                                        HOSTS_FIELDS_V1_URL,
@@ -427,13 +431,13 @@ class SatelliteSixV1(SatelliteInterface):
                                               org_id, host_id))
             logger.debug('host_id=%s, host_details=%s',
                          host_id, details)
-            return {'host_name': host_name,
+            return {'name': unique_name,
                     'details': details,
                     'status': SystemInspectionResult.SUCCESS}
         except SatelliteException as sat_error:
             error_message = 'Satellite error encountered: %s\n' % sat_error
             logger.error(error_message)
-            return {'host_name': host_name,
+            return {'name': unique_name,
                     'details': details,
                     'status': SystemInspectionResult.FAILED}
         return details
@@ -488,9 +492,10 @@ class SatelliteSixV1(SatelliteInterface):
                         for result in results:
                             if result is not None:
                                 self.record_inspect_result(
-                                    result.get('host_name'),
+                                    result.get('name'),
                                     result.get('details'),
                                     result.get('status'))
+        utils.validate_task_stats(self.inspect_scan_task)
 
 
 class SatelliteSixV2(SatelliteInterface):
@@ -532,10 +537,13 @@ class SatelliteSixV2(SatelliteInterface):
                                          (response.status_code, url))
             jsonresult = response.json()
             for result in jsonresult.get(RESULTS, []):
-                name = result.get(NAME)
-                if name is not None:
-                    hosts.append(name)
-                    self.record_conn_result(name, credential)
+                host_name = result.get(NAME)
+                host_id = result.get(ID)
+
+                if host_name is not None and host_id is not None:
+                    unique_name = '%s_%s' % (host_name, host_id)
+                    hosts.append(unique_name)
+                    self.record_conn_result(unique_name, credential)
 
         return hosts
 
@@ -550,15 +558,16 @@ class SatelliteSixV2(SatelliteInterface):
             raise SatelliteException(
                 'host_details cannot be called for a connection scan')
         details = {}
+        unique_name = '%s_%s' % (host_name, host_id)
         sys_result = self.inspect_scan_task.inspection_result.systems.filter(
-            name=host_name).first()
+            name=unique_name).first()
 
         if sys_result:
             logger.debug('Results already captured for host_name=%s',
                          host_name)
             return None
         try:
-            message = 'REQUESTING HOST DETAILS: %s' % host_name
+            message = 'REQUESTING HOST DETAILS: %s' % unique_name
             self.inspect_scan_task.log_message(message)
             details.update(host_fields(self.inspect_scan_task, 2,
                                        HOSTS_FIELDS_V2_URL,
@@ -568,13 +577,13 @@ class SatelliteSixV2(SatelliteInterface):
                                               None, host_id))
             logger.debug('host_id=%s, host_details=%s',
                          host_id, details)
-            return {'host_name': host_name,
+            return {'name': unique_name,
                     'details': details,
                     'status': SystemInspectionResult.SUCCESS}
         except SatelliteException as sat_error:
             error_message = 'Satellite error encountered: %s\n' % sat_error
             logger.error(error_message)
-            return {'host_name': host_name,
+            return {'name': unique_name,
                     'details': details,
                     'status': SystemInspectionResult.FAILED}
         return details
@@ -623,6 +632,8 @@ class SatelliteSixV2(SatelliteInterface):
                     for result in results:
                         if result is not None:
                             self.record_inspect_result(
-                                result.get('host_name'),
+                                result.get('name'),
                                 result.get('details'),
                                 result.get('status'))
+
+        utils.validate_task_stats(self.inspect_scan_task)

--- a/quipucords/scanner/satellite/tests_sat_five.py
+++ b/quipucords/scanner/satellite/tests_sat_five.py
@@ -17,7 +17,9 @@ from api.models import (Credential,
                         ScanJob,
                         ScanTask,
                         Source,
-                        SystemInspectionResult)
+                        SystemConnectionResult,
+                        SystemInspectionResult,
+                        TaskConnectionResult)
 
 from django.test import TestCase
 
@@ -60,6 +62,18 @@ class SatelliteFiveTest(TestCase):
             self.source, ScanTask.SCAN_TYPE_INSPECT)
 
         self.api = SatelliteFive(self.scan_job, self.scan_task)
+        connection_results = TaskConnectionResult()
+        connection_results.save()
+        self.api.connect_scan_task.connection_result = connection_results
+        self.api.connect_scan_task.connection_result.save()
+
+        sys_result = SystemConnectionResult(
+            name='sys1_1',
+            status=SystemInspectionResult.SUCCESS)
+        sys_result.save()
+        self.api.connect_scan_task.connection_result.systems.add(sys_result)
+        self.api.connect_scan_task.connection_result.save()
+        self.api.connect_scan_task.save()
 
     def tearDown(self):
         """Cleanup test case setup."""
@@ -86,9 +100,12 @@ class SatelliteFiveTest(TestCase):
     @patch('xmlrpc.client.ServerProxy')
     def test_hosts(self, mock_serverproxy):
         """Test the method hosts."""
-        systems = [{'name': 'sys1'},
-                   {'name': 'sys2'},
-                   {'name': 'sys3'}]
+        systems = [{'name': 'sys1',
+                    'id': 1},
+                   {'name': 'sys2',
+                    'id': 2},
+                   {'name': 'sys3',
+                    'id': 3}]
         client = mock_serverproxy.return_value
         client.auth.login.return_value = 'key'
         client.auth.logout.return_value = 'key'
@@ -97,7 +114,7 @@ class SatelliteFiveTest(TestCase):
         hosts = self.api.hosts()
         self.assertEqual(systems_count, 3)
         self.assertEqual(len(hosts), 3)
-        self.assertEqual(hosts, ['sys1', 'sys2', 'sys3'])
+        self.assertEqual(hosts, ['sys1_1', 'sys2_2', 'sys3_3'])
 
     @patch('xmlrpc.client.ServerProxy')
     def test_hosts_with_err(self, mock_serverproxy):
@@ -137,7 +154,7 @@ class SatelliteFiveTest(TestCase):
                                         last_checkin='', virtual_hosts=virt,
                                         virtual_guests={}, physical_hosts=[])
         self.assertEqual(details, {'details': expected,
-                                   'host_name': 'sys1',
+                                   'name': 'sys1_1',
                                    'status': 'success'})
 
     @patch('xmlrpc.client.ServerProxy')
@@ -172,14 +189,14 @@ class SatelliteFiveTest(TestCase):
                                         virtual_guests={1: 2},
                                         physical_hosts=[])
         self.assertEqual(details, {'details': expected,
-                                   'host_name': 'sys1',
+                                   'name': 'sys1_1',
                                    'status': 'success'})
 
     def test_host_details_skip(self):
         """Test host_details method for already captured data."""
         # pylint: disable=no-member
         sys_result = SystemInspectionResult(
-            name='sys1',
+            name='sys1_1',
             status=SystemInspectionResult.SUCCESS)
         sys_result.save()
         inspect_result = self.scan_task.inspection_result
@@ -208,7 +225,7 @@ class SatelliteFiveTest(TestCase):
         self.assertEqual(len(inspect_result.systems.all()), 0)
         self.assertEqual(
             details, {'details': {},
-                      'host_name': 'sys1',
+                      'name': 'sys1_1',
                       'status': 'failed'})
 
     @patch('xmlrpc.client.ServerProxy')
@@ -284,7 +301,7 @@ class SatelliteFiveTest(TestCase):
             self.api.hosts_facts(Value('i', ScanJob.JOB_RUN))
 
     @patch('multiprocessing.pool.Pool.starmap', return_value=[
-        {'host_name': 'sys10',
+        {'name': 'sys10',
          'details': {},
          'status': SystemInspectionResult.SUCCESS}])
     @patch('xmlrpc.client.ServerProxy')

--- a/quipucords/scanner/satellite/tests_sat_six.py
+++ b/quipucords/scanner/satellite/tests_sat_six.py
@@ -17,7 +17,9 @@ from api.models import (Credential,
                         ScanOptions,
                         ScanTask,
                         Source,
-                        SystemInspectionResult)
+                        SystemConnectionResult,
+                        SystemInspectionResult,
+                        TaskConnectionResult)
 
 from django.test import TestCase
 
@@ -65,6 +67,18 @@ class SatelliteSixV1Test(TestCase):
             self.source, ScanTask.SCAN_TYPE_INSPECT)
 
         self.api = SatelliteSixV1(self.scan_job, self.scan_task)
+        connection_results = TaskConnectionResult()
+        connection_results.save()
+        self.api.connect_scan_task.connection_result = connection_results
+        self.api.connect_scan_task.connection_result.save()
+
+        sys_result = SystemConnectionResult(
+            name='sys1_1',
+            status=SystemInspectionResult.SUCCESS)
+        sys_result.save()
+        self.api.connect_scan_task.connection_result.systems.add(sys_result)
+        self.api.connect_scan_task.connection_result.save()
+        self.api.connect_scan_task.save()
 
     def tearDown(self):
         """Cleanup test case setup."""
@@ -136,9 +150,12 @@ class SatelliteSixV1Test(TestCase):
             '/v2/organizations/{org_id}/systems'
         with requests_mock.Mocker() as mocker:
             url = construct_url(url=hosts_url, sat_host='1.2.3.4', org_id=1)
-            jsonresult = {'results': [{'name': 'sys1'},
-                                      {'name': 'sys2'},
-                                      {'name': 'sys3'}],
+            jsonresult = {'results': [{'name': 'sys1',
+                                       'id': 1},
+                                      {'name': 'sys2',
+                                       'id': 2},
+                                      {'name': 'sys3',
+                                       'id': 3}],
                           'per_page': 100,
                           'total': 3}
             mocker.get(url, status_code=200, json=jsonresult)
@@ -146,7 +163,7 @@ class SatelliteSixV1Test(TestCase):
             hosts = self.api.hosts()
             self.assertEqual(systems_count, 3)
             self.assertEqual(len(hosts), 3)
-            self.assertEqual(hosts, ['sys1', 'sys2', 'sys3'])
+            self.assertEqual(hosts, ['sys1_1', 'sys2_2', 'sys3_3'])
 
     @patch('scanner.satellite.six.SatelliteSixV1.get_orgs')
     def test_hosts_with_err(self, mock_get_orgs):
@@ -235,7 +252,7 @@ class SatelliteSixV1Test(TestCase):
         """Test host_details method for already captured data."""
         # pylint: disable=no-member
         sys_result = SystemInspectionResult(
-            name='sys1',
+            name='sys1_1',
             status=SystemInspectionResult.SUCCESS)
         sys_result.save()
         inspect_result = self.scan_task.inspection_result
@@ -253,7 +270,7 @@ class SatelliteSixV1Test(TestCase):
             inspect_result = self.scan_task.inspection_result
             self.assertEqual(len(inspect_result.systems.all()), 0)
             self.assertEqual(
-                detail, {'host_name': 'sys1',
+                detail, {'name': 'sys1_1',
                          'details': {},
                          'status': 'failed'})
             mock_fields.assert_called_once_with(ANY, ANY, ANY, ANY, ANY)
@@ -346,7 +363,7 @@ class SatelliteSixV1Test(TestCase):
                                                 host_id=1,
                                                 host_name='sys1')
                 self.assertEqual(
-                    details, {'host_name': 'sys1',
+                    details, {'name': 'sys1_1',
                               'details': expected,
                               'status': 'success'})
                 mock_fields.assert_called_once_with(ANY, ANY, ANY, ANY, ANY)
@@ -365,7 +382,7 @@ class SatelliteSixV1Test(TestCase):
                 self.api.hosts_facts(Value('i', ScanJob.JOB_RUN))
 
     @patch('multiprocessing.pool.Pool.starmap', return_value=[
-        {'host_name': 'sys10',
+        {'name': 'sys10',
          'details': {},
          'status': SystemInspectionResult.SUCCESS}])
     def test_hosts_facts(self, mock_pool):
@@ -420,6 +437,18 @@ class SatelliteSixV2Test(TestCase):
             self.source, ScanTask.SCAN_TYPE_INSPECT)
         self.scan_task.update_stats('TEST_SAT.', sys_scanned=0)
         self.api = SatelliteSixV2(self.scan_job, self.scan_task)
+        connection_results = TaskConnectionResult()
+        connection_results.save()
+        self.api.connect_scan_task.connection_result = connection_results
+        self.api.connect_scan_task.connection_result.save()
+
+        sys_result = SystemConnectionResult(
+            name='sys1_1',
+            status=SystemInspectionResult.SUCCESS)
+        sys_result.save()
+        self.api.connect_scan_task.connection_result.systems.add(sys_result)
+        self.api.connect_scan_task.connection_result.save()
+        self.api.connect_scan_task.save()
 
     def tearDown(self):
         """Cleanup test case setup."""
@@ -458,9 +487,12 @@ class SatelliteSixV2Test(TestCase):
         hosts_url = 'https://{sat_host}:{port}/api/v2/hosts'
         with requests_mock.Mocker() as mocker:
             url = construct_url(url=hosts_url, sat_host='1.2.3.4', org_id=1)
-            jsonresult = {'results': [{'name': 'sys1'},
-                                      {'name': 'sys2'},
-                                      {'name': 'sys3'}],
+            jsonresult = {'results': [{'name': 'sys1',
+                                       'id': 1},
+                                      {'name': 'sys2',
+                                       'id': 2},
+                                      {'name': 'sys3',
+                                       'id': 3}],
                           'per_page': 100,
                           'total': 3}
             mocker.get(url, status_code=200, json=jsonresult)
@@ -468,7 +500,7 @@ class SatelliteSixV2Test(TestCase):
             hosts = self.api.hosts()
             self.assertEqual(systems_count, 3)
             self.assertEqual(len(hosts), 3)
-            self.assertEqual(hosts, ['sys1', 'sys2', 'sys3'])
+            self.assertEqual(hosts, ['sys1_1', 'sys2_2', 'sys3_3'])
 
     def test_hosts_with_err(self):
         """Test the method hosts with error."""
@@ -654,7 +686,7 @@ class SatelliteSixV2Test(TestCase):
             inspect_result = self.scan_task.inspection_result
             self.assertEqual(len(inspect_result.systems.all()), 0)
             self.assertEqual(
-                detail, {'host_name': 'sys1',
+                detail, {'name': 'sys1_1',
                          'details': {},
                          'status': 'failed'})
             mock_fields.assert_called_once_with(ANY, ANY, ANY, ANY, ANY)
@@ -742,7 +774,7 @@ class SatelliteSixV2Test(TestCase):
                        return_value=subs_return_value) as mock_subs:
                 details = self.api.host_details(host_id=1, host_name='sys1')
                 self.assertEqual(
-                    details, {'host_name': 'sys1',
+                    details, {'name': 'sys1_1',
                               'details': expected,
                               'status': 'success'})
                 mock_fields.assert_called_once_with(ANY, ANY, ANY, ANY, ANY)
@@ -752,7 +784,7 @@ class SatelliteSixV2Test(TestCase):
         """Test host_details method for already captured data."""
         # pylint: disable=no-member
         sys_result = SystemInspectionResult(
-            name='sys1',
+            name='sys1_1',
             status=SystemInspectionResult.SUCCESS)
         sys_result.save()
         inspect_result = self.scan_task.inspection_result
@@ -772,7 +804,7 @@ class SatelliteSixV2Test(TestCase):
                 self.api.hosts_facts(Value('i', ScanJob.JOB_RUN))
 
     @patch('multiprocessing.pool.Pool.starmap', return_value=[
-        {'host_name': 'sys10',
+        {'name': 'sys10',
          'details': {},
          'status': SystemInspectionResult.SUCCESS}])
     def test_hosts_facts(self, mock_pool):
@@ -785,6 +817,18 @@ class SatelliteSixV2Test(TestCase):
             scan_options=scan_options)
         scan_task.update_stats('TEST_SAT.', sys_scanned=0)
         api = SatelliteSixV2(scan_job, scan_task)
+        connection_results = TaskConnectionResult()
+        connection_results.save()
+        api.connect_scan_task.connection_result = connection_results
+        api.connect_scan_task.connection_result.save()
+
+        sys_result = SystemConnectionResult(
+            name='sys1_1',
+            status=SystemInspectionResult.SUCCESS)
+        sys_result.save()
+        api.connect_scan_task.connection_result.systems.add(sys_result)
+        api.connect_scan_task.connection_result.save()
+        api.connect_scan_task.save()
         hosts_url = 'https://{sat_host}:{port}/api/v2/hosts'
         with requests_mock.Mocker() as mocker:
             url = construct_url(url=hosts_url, sat_host='1.2.3.4')

--- a/quipucords/scanner/satellite/tests_sat_utils.py
+++ b/quipucords/scanner/satellite/tests_sat_utils.py
@@ -33,7 +33,8 @@ from scanner.satellite.utils import (_status5,
                                      get_connect_data,
                                      get_credential,
                                      get_sat5_client,
-                                     status)
+                                     status,
+                                     validate_task_stats)
 from scanner.test_util import create_scan_job
 
 
@@ -193,3 +194,13 @@ class SatelliteUtilsTest(TestCase):
         self.assertIsNotNone(client)
         self.assertEqual(user, 'username')
         self.assertEqual(password, 'password')
+
+    def test_validate_task_stats(self):
+        """Test validate task stats no errors."""
+        validate_task_stats(self.scan_task)
+
+    def test_validate_task_stats_error(self):
+        """Test validate task stats errors."""
+        with self.assertRaises(SatelliteException):
+            self.scan_task.increment_stats('TEST', increment_sys_count=True)
+            validate_task_stats(self.scan_task)

--- a/quipucords/scanner/satellite/utils.py
+++ b/quipucords/scanner/satellite/utils.py
@@ -200,3 +200,25 @@ def data_map(mapping_dict, data):
         for key, mapping_key in mapping_dict.items():
             out[key] = data.get(mapping_key)
     return out
+
+
+def validate_task_stats(task):
+    """Map data keys to new output.
+
+    :param task: ScanTask to evaluate
+    :throws: SatelliteException if task stats are not valid
+    """
+    systems_count,\
+        systems_scanned, \
+        systems_failed, \
+        systems_unreachable = task.calculate_counts()
+    totals = + systems_scanned + systems_failed + systems_unreachable
+    if totals != systems_count:
+        missing_sys = systems_count - totals
+        error = 'Scan failed to inspect %d systems.' % missing_sys
+        task.log_message(error, log_level=logging.ERROR)
+        new_failed = missing_sys + systems_failed
+        task.update_stats(
+            'Missed failed systems', sys_failed=new_failed)
+        raise SatelliteException(
+            'hosts_facts could not scan all systems')

